cl_hubeau/water_services/utils.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Convenience functions for water services indicators.
+"""
+
+import geopandas as gpd
+import pandas as pd
+from tqdm import tqdm
+
+from cl_hubeau.water_services.water_services_scraper import WaterServicesSession
+from cl_hubeau import _config
+from cl_hubeau.utils import get_departements_from_regions, get_regions, prepare_kwargs_loops
+from datetime import datetime
+
+def get_all_communes(**kwargs) -> gpd.GeoDataFrame:
+    """
+    Gets the services for every french city.
+
+    Parameters
+    ----------
+    **kwargs :
+        kwargs passed to WaterServicesSession.get_communes (hence mostly intended
+        for hub'eau API's arguments). Do not use `format` or `code_commune`
+        as they are set by the current function.
+
+    Returns
+    -------
+    results : gpd.GeoDataFrame
+        GeoDataFrame of water services indicators for every french city.
+
+    """
+    with WaterServicesSession() as session :
+        regions = get_regions(latest=True)
+        results = [
+            session.get_communes(
+                code_departement=get_departements_from_regions(reg), detail_service=True, format="geojson", **kwargs
```

Comment:
This won't work: `get_departements_from_regions` returns a list and `get_communes` expects a single str.

Maybe we should proceed as following:
* add a `code_region` to `get_all_communes` arguments, with None as default ;
* then, outside of loop, compute the departements codes if the user has either set `code_region` or `code_departement`;
* if the user set `code_commune` altogether, only one query will be needed;
* prepare the kwargs to pass to `get_communes` inside the loop (either a list of one dict or a list of multiple dicts, depending of the two previous cases)
* the execute the loop